### DISCUSSION TITLE
Suggestions about red life, dizzy points and guard points

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#918] [Feature Request] [answered]
As requested, I'm posting these suggestions over here for visibility.

The way red life, dizzy points and guard points currently work is a bit counterintuitive compared to how regular damage works. For instance, when you write a Hitdef, you have to use positive values for damage, positive values for red life, then negative values for dizzy and guard points. I think they should all be handled based on how regular damage works. That is, positive values for everything (while negative values would "heal" the respective bars).

This also means that the constants that handle default behaviour for those values should be positive instead of negative.

On the other hand, I think all state controllers that add to those values should be modeled after LifeAdd/TargetLifeAdd when it comes to that. That is negative values subtract, while positive values add to the bars.

The other thing is that the way red life is set up right now works, but could be more intuitive and closer to how commercial games handle it. In Capcom games, the red life is like a shadow of the actual life. The character starts the round with both life and red life full (1000 points each for instance), then as it receives damage both values decrease, with red life decreasing more slowly. When the character is tagged out, actual life recovers until it is the same as red life.

As for how this would be handled in Hitdef, I'm not sure myself. But for consistency's sake, and if you consider the idea that RedLife is a shadow of Life, I think it should all be modeled after Damage and LifeAdd again. So a value of 0 in RedLife would mean that the damage is 100% recoverable. This means that "Default.LifeToRedLifeMul" would work the opposite of how it currently does, with a value of 0.25 making 75% of damage recoverable.

It's all a bit confusing, so anything let me know.

**Comment by K4thos** [comment, marked as answer]
Marking as answered since I believe this has already been implemented by you
